### PR TITLE
rack: attach more debugging information

### PR DESCRIPTION
**Pull request by kyrylo**

### CODE REVIEW

**Comment by kyrylo** [comment]
PTAL

File: spec/integration/shared_examples/rack_examples.rb [hunk 1]
```diff
@@ -101,6 +101,18 @@ def wait_for_a_request_with_body(body)
       it "features userAgent" do
         wait_for_a_request_with_body(/"context":{.*"userAgent":"Bot".*}/)
       end
+
+      it "features referer" do
+        wait_for_a_request_with_body(/"context":{.*"referer":"bingo.com".*}/)
+      end
+
+      it "contains HTTP headers" do
```
Comment (zefer) [inline spec/integration/shared_examples/rack_examples.rb]: Should we also include a test for `CONTENT_TYPE`?

**Comment by zefer** [comment]
LGTM

File: spec/integration/shared_examples/rack_examples.rb (hunk 1 above)
Comment (kyrylo) [inline spec/integration/shared_examples/rack_examples.rb]: This is an example for an error in someone else's app. We simply ensure here that we are able to report headers if there were any. We already check our own content type when we send a request to our server. We have that test in airbrake-ruby.

File: lib/airbrake/rack/notice_builder.rb [hunk 2]
```diff
@@ -45,6 +54,9 @@ def add_context(notice)
 
         context[:url] = @request.url
         context[:userAgent] = @request.user_agent
+        context[:httpMethod] = @request.request_method
+        context[:referer] = @request.referer
+        context[:headers] = request_headers
```
Comment (vmihailenco) [inline lib/airbrake/rack/notice_builder.rb]: Airbrake API is free to throw away such properties, because they are not officially supported.

Comment (kyrylo) [inline lib/airbrake/rack/notice_builder.rb]: Let's support them.

**Pull request merged**